lib/new.js

```diff
@@ -0,0 +1,687 @@
+/**
+ * Copyright 2018 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const path = require('path');
+const puppeteer = require('puppeteer-core');const findChrome = require('./find_chrome');
+const {rpc} = require('../rpc');
+const debugApp = require('debug')('carlo:app');
+const debugServer = require('debug')('carlo:server');
```

Comment:
format: style mismatch:
```javascript
22|const debugApp = require('debug')('carlo:app');
23|const debugServer = require('debug')('carlo:server');
24|const {Color} = require('./color');
```
```suggestion
const debugServer = require('debug')('carlo:server');const {Color} = require('./color');const {HttpRequest} = require('./http_request');const fs = require('fs');const util = require('util');const {URL} = require('url');const EventEmitter = require('events');const fsReadFile = util.promisify(fs.readFile);let testMode = false;class App extends EventEmitter {/**
```

⏎ at column 54 should be removed.
Triggered rule # 279
```
  +1.roles not in {ARGUMENT, CALLEE}
	∧ +1.reserved not in {=, {}
	∧ +2.length ≥ 3
	∧ +2.reserved not in {(}
	∧ +2.roles not in {RIGHT}
	∧ +3.roles not in {EXPRESSION}
	∧ +5.internal_type not in {Identifier}
	∧ ^1.roles in {BLOCK} and not in {IDENTIFIER, OPERATOR, DECLARATION}
	∧ ^2.roles not in {CALL}
	∧ -1.roles not in {STRING, EXPRESSION}
	∧ -1.reserved not in {,, :, ), .}
	∧ -1.diff_offset ≤ 1
	∧ -2.diff_line = 0
	∧ -2.label not in {<-space>}
	∧ -2.diff_col ≤ 3
	∧ -3.label not in {<newline>}
	∧ -4.roles not in {CALL}
	∧ -4.reserved not in {;}
	∧ •••start_col ≥ 16
	⇒ y = ⏎
	Confidence: 0.782. Support: 62.
```
⏎ at column 36 should be removed.
Triggered rule # 143
```
  +1.roles in {MAP} and not in {ARGUMENT}
	∧ +2.roles not in {RIGHT}
	∧ ^1.roles not in {DECLARATION}
	∧ ^2.roles not in {BINARY, MAP}
	∧ -1.roles not in {STRING, EXPRESSION}
	∧ -1.reserved = , and not in {:, .}
	∧ -2.diff_line = 0
	∧ -3.roles not in {VALUE}
	∧ -4.roles in {LITERAL}
	⇒ y = ⏎
	Confidence: 0.812. Support: 8.
```
⏎⏎ at column 49 should be removed.
Triggered rule # 121
```
  +1.roles in {ARGUMENT}
	∧ +2.roles not in {RIGHT}
	∧ +4.reserved = const
	∧ ^1.roles not in {DECLARATION}
	∧ -1.roles not in {STRING, EXPRESSION}
	∧ -1.reserved not in {:, .}
	∧ -2.diff_line = 0
	∧ -3.length ≤ 8
	∧ -3.diff_offset ≥ 4
	∧ -3.label in {<space>}
	⇒ y = '
	Confidence: 0.976. Support: 21.
```
⏎ at column 26 should be removed.
Triggered rule # 274
```
  +1.roles not in {ARGUMENT, CALLEE}
	∧ +1.reserved not in {=, {}
	∧ +2.roles not in {RIGHT}
	∧ +2.reserved = (
	∧ ^1.roles in {BLOCK} and not in {IDENTIFIER, OPERATOR, DECLARATION}
	∧ ^2.roles not in {CALL}
	∧ -1.roles not in {STRING, EXPRESSION}
	∧ -1.reserved not in {,, :, ), .}
	∧ -1.diff_offset ≤ 1
	∧ -2.diff_line = 0
	∧ -2.label not in {<-space>}
	∧ -2.diff_col ≤ 3
	∧ -3.label not in {<newline>}
	∧ -4.roles not in {CALL}
	∧ -4.reserved not in {;}
	⇒ y = ⏎
	Confidence: 0.837. Support: 46.
```
⏎ at column 30 should be removed.
Triggered rule # 278
```
  +1.roles not in {ARGUMENT, CALLEE}
	∧ +1.reserved not in {=, {}
	∧ +2.length ≥ 3
	∧ +2.reserved not in {(}
	∧ +2.roles not in {RIGHT}
	∧ +3.roles in {EXPRESSION}
	∧ +5.internal_type not in {Identifier}
	∧ ^1.roles in {BLOCK} and not in {IDENTIFIER, OPERATOR, DECLARATION}
	∧ ^2.roles not in {CALL}
	∧ -1.roles not in {STRING, EXPRESSION}
	∧ -1.reserved not in {,, :, ), .}
	∧ -1.diff_offset ≤ 1
	∧ -2.diff_line = 0
	∧ -2.label not in {<-space>}
	∧ -2.diff_col ≤ 3
	∧ -3.label not in {<newline>}
	∧ -4.roles not in {CALL}
	∧ -4.reserved not in {;}
	⇒ y = ∅
	Confidence: 0.812. Support: 8.
```
⏎ at column 30 should be removed.
Triggered rule # 81
```
  +2.roles not in {RIGHT, CALL}
	∧ +5.reserved = {
	∧ ^1.roles in {DECLARATION}
	∧ ^1.internal_type not in {ClassDeclaration}
	∧ -1.length ≤ 1
	∧ -1.reserved not in {:, .}
	∧ -1.roles not in {STRING, EXPRESSION}
	∧ -3.diff_col ≥ 4
	∧ -4.roles not in {NAME}
	∧ -4.reserved not in {,}
	∧ -4.diff_offset ≤ 12
	∧ -5.reserved not in {const}
	∧ •••start_col ≥ 8
	⇒ y = ␣
	Confidence: 0.883. Support: 30.
```
⏎ at column 40 should be removed.
Triggered rule # 111
```
  +1.roles in {EXPRESSION} and not in {LITERAL}
	∧ +2.roles not in {RIGHT}
	∧ +3.roles in {EXPRESSION} and not in {CALLEE}
	∧ +5.internal_type not in {Identifier}
	∧ ^1.roles not in {SCOPE, DECLARATION}
	∧ ^2.internal_type not in {File}
	∧ -1.length ≤ 2
	∧ -1.reserved not in {:, .}
	∧ -1.roles not in {STRING, EXPRESSION}
	∧ -2.diff_line ≥ 1
	∧ -5.diff_line ≥ 2
	∧ •••start_line ≥ 34
	⇒ y = ∅
	Confidence: 0.962. Support: 13.
```
⏎⏎ at column 48 should be removed.
Triggered rule # 298
```
  +1.roles not in {ARGUMENT, CALLEE, STRING}
	∧ +1.reserved not in {=, {}
	∧ +2.roles not in {RIGHT}
	∧ +3.roles not in {COMMENT}
	∧ +3.reserved not in {;}
	∧ +4.roles not in {IDENTIFIER}
	∧ ^1.roles not in {BLOCK, IDENTIFIER, OPERATOR, DECLARATION}
	∧ ^2.internal_type not in {ClassDeclaration}
	∧ -1.roles not in {STRING, EXPRESSION}
	∧ -1.reserved not in {,, :, ), .}
	∧ -2.diff_line = 0
	∧ -2.reserved = )
	∧ -2.label not in {<-space>}
	∧ -2.diff_col ≤ 3
	∧ -3.label not in {<newline>}
	∧ -4.roles not in {CALL, EXPRESSION}
	∧ -4.reserved not in {;}
	∧ -4.internal_type not in {NumericLiteral}
	∧ -4.diff_col ≤ 10
	∧ -5.diff_offset ≤ 37
	⇒ y = ⏎␣⁻␣⁻
	Confidence: 0.295. Support: 22.
```
⏎⏎ at column 22 should be removed.
Triggered rule # 298
```
  +1.roles not in {ARGUMENT, CALLEE, STRING}
	∧ +1.reserved not in {=, {}
	∧ +2.roles not in {RIGHT}
	∧ +3.roles not in {COMMENT}
	∧ +3.reserved not in {;}
	∧ +4.roles not in {IDENTIFIER}
	∧ ^1.roles not in {BLOCK, IDENTIFIER, OPERATOR, DECLARATION}
	∧ ^2.internal_type not in {ClassDeclaration}
	∧ -1.roles not in {STRING, EXPRESSION}
	∧ -1.reserved not in {,, :, ), .}
	∧ -2.diff_line = 0
	∧ -2.reserved = )
	∧ -2.label not in {<-space>}
	∧ -2.diff_col ≤ 3
	∧ -3.label not in {<newline>}
	∧ -4.roles not in {CALL, EXPRESSION}
	∧ -4.reserved not in {;}
	∧ -4.internal_type not in {NumericLiteral}
	∧ -4.diff_col ≤ 10
	∧ -5.diff_offset ≤ 37
	⇒ y = ⏎␣⁻␣⁻
	Confidence: 0.295. Support: 22.
```
⏎␣⁺␣⁺ at column 33 should be removed.
Triggered rule # 282
```
  +1.roles not in {ARGUMENT, CALLEE, THIS}
	∧ +1.reserved not in {=, {}
	∧ +2.length ≤ 2
	∧ +2.reserved not in {(}
	∧ +2.roles not in {RIGHT}
	∧ ^1.roles in {BLOCK} and not in {IDENTIFIER, OPERATOR, DECLARATION}
	∧ ^2.roles not in {CALL}
	∧ -1.roles not in {STRING, EXPRESSION}
	∧ -1.reserved not in {,, :, ), .}
	∧ -1.diff_offset ≤ 1
	∧ -2.diff_line = 0
	∧ -2.label not in {<-space>}
	∧ -2.diff_col ≤ 3
	∧ -3.roles in {IDENTIFIER}
	∧ -3.label not in {<newline>}
	∧ -4.roles not in {CALL}
	∧ -4.reserved not in {;}
	∧ -4.diff_col ≥ 6
	∧ •••start_col ≥ 18
	⇒ y = ∅
	Confidence: 0.690. Support: 21.
```